File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        List<List<Integer>> v = new ArrayList<List<Integer>>();
        if (root == null) {
            return v;
        }
        q.add(root);
        while (!q.isEmpty()) {
            int qsize = q.size();
            List<Integer> subv = new ArrayList<>();
            for (int i = 0; i < qsize; i++) {
                if (q.peek().left != null) {
                    q.add(q.peek().left);
                }
                if (q.peek().right != null) {
                    q.add(q.peek().right);
                }
                subv.add(q.poll().val);
            }
            v.add(subv);
        }
        return v;
    }
}

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode*> q;
        vector<vector<int>> ans;
        q.push(root);
        loT(root, q, ans);

        return ans;
    }
    void loT(TreeNode* node, queue<TreeNode*>& q, vector<vector<int>>& ans) {
        if (node == nullptr) {
            return;
        }
        while (!q.empty()) {
            vector<int> tmp;
            int n = q.size();
            for (int i = 0; i < n; i++) {
                TreeNode* frontNode = q.front();
                int front = frontNode->val;
                q.pop();
                tmp.push_back(front);
                if (frontNode->left != nullptr) {
                    q.push(frontNode->left);
                }
                if (frontNode->right != nullptr) {
                    q.push(frontNode->right);
                }
            }
            ans.push_back(tmp);
        }
    }
};